src/main/java/com/gooddata/executeafm/afm/CopyWithUriQualifierFactory.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2007-2018, GoodData(R) Corporation. All rights reserved.
+ */
+package com.gooddata.executeafm.afm;
+
+import com.gooddata.executeafm.UriObjQualifier;
+
+/**
+ * The interface of the function that defines the factory that can copy any object with provided {@link UriObjQualifier}.
+ *
+ * @param <T>
+ *         The type of the copied object.
+ */
+@FunctionalInterface
+interface CopyWithUriQualifierFactory<T> {
```

Comment:
I would remove this interface, since it's inherently pubůloishes something whis is package private otherwise, just use the common fucntion in the utility

Reply:
Refactored to the generic `Function` as suggested.